task/rhacs-generic/0.1/README.md

```diff
@@ -0,0 +1,110 @@
+
+<https://www.redhat.com/en/technologies/cloud-computing/openshift/advanced-cluster-security-kubernetes>
+
+## Install the Task
+
+```bash
+kubectl apply -f https://api.hub.tekton.dev/v1/resource/tekton/task/rhacs-generic/0.1/raw
+```
+
+## Parameters
+
+- **`rox_central_endpoint`**: The address:port tuple for StackRox Central.
+Default: **central.stackrox.svc:443**
+- **`insecure-skip-tls-verify`**: Skip verification the TLS certs of the Central
+endpoint and registry. Examples: _"true", **"false"**_.
+- **`rox_arguments`**: The command line that you would like to enter, in array
+form. (required)
+- `rox_image`: The image providing the roxctl tool (optional).
+Default: quay.io/stackrox-io/roxctl:4.4.2
+(this is also the minimum version working with this task). 
+- `output_file`: path to a file where to redirect roxctl standard output.
+Default: "" (redirects to stdout).
+- `error_file`: path to a file where to redirect roxctl standard error.
+Default: "" (redirects to stderr).
+
+## Workspaces
+
+- **data**: An [optional Workspace](https://github.com/tektoncd/pipeline/blob/main/docs/workspaces.md#optional-workspaces)
+which stores files used as parameters to the command line.
+
+## Usage
+
+Check the [documentation](https://docs.openshift.com/acs/operating/manage-user-access/configure-short-lived-access.html#configure-short-lived-access_configure-short-lived-access)
+to configure the trust with the OIDC token issuer. This
+[example](../../rhacs-m2m-authenticate/0.1/samples/configure-m2m.md) describes
+a possible RHACS machine-to-machine integration configuration.
+
+**Example task uses:**
+
+Deployment check:
+```yaml
+  tasks:
+    - name: check-deployment
+      taskRef:
+        name: rhacs-generic
+        kind: Task
+      params:
+        - name: insecure-skip-tls-verify
+          value: "true"
+        - name: rox_endpoint
+          value: central.stackrox.svc:443
+        - name: rox_arguments
+          value:
+            - deployment
+            - check
+            - --output=table
+            - --file=/workspace/data/$(params.deployment)
+      workspaces:
+        - name: data
+          workspace: shared-workspace
+```
+
+Image scan:
+```yaml
+  tasks:
+    - name: scan-image
+      taskRef:
+        name: rhacs-generic
+        kind: Task
+      params:
+        - name: insecure-skip-tls-verify
+          value: "true"
+        - name: rox_endpoint
+          value: central.stackrox.svc:443
+        - name: rox_arguments
+          value:
+            - image
+            - scan
+            - --output=table
+            - --image=$(params.IMAGE)@$(tasks.build-image.results.IMAGE_DIGEST)
+      runAfter:
+        - build-image
+
+```
+
+**Samples:**
+
+* [pipeline.yaml](samples/pipeline.yaml) demonstrates use in a pipeline.
+* [pipelinerun.yaml](samples/pipelinerun.yaml) demonstrates use
+in a pipelinerun.
+
+# Known Issues
+
+* Skipping TLS Verify is currently required. TLS trust bundle not working for
+quay.io etc.
```

Comment:
I know we have this in the other tasks but: is this still required?

Reply:
If the option is left aside, the TaskRun issues this warning:
```
WARN:	The remote endpoint failed TLS validation. This will be a fatal error in future releases.
Please do one of the following at your earliest convenience:
  1. Obtain a valid certificate for your Central instance/Load Balancer.
  2. Use the --ca option to specify a custom CA certificate (PEM format). This Certificate can be obtained by
     running "roxctl central cert".
  3. Update all your roxctl usages to pass the --insecure-skip-tls-verify option, in order to
     suppress this warning and retain the old behavior of not validating TLS certificates in
     the future (NOT RECOMMENDED).


WARN:	Certificate validation error: x509: certificate signed by unknown authority
```

Reply:
The certificate authority could be passed to the generic task similarly to what was done for the API Token/Short-lived token in the specific action tasks.
I'll rework the task to do that.